src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,11 +1,40 @@
 package core.basesyntax.service;
 
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMUM_USER_AGE = 18;
+    private static final int MINIMUM_PASSWORD_LENGTH = 6;
+    private static final int MINIMUM_LOGIN_LENGTH = 1;
 
     @Override
     public User register(User user) {
-        return null;
+        StorageDao storageDao = new StorageDaoImpl();
+        validateUserData(user);
+        checkIsLoginAlreadyRegistered(user.getLogin());
+        return storageDao.add(user);
+    }
+
+    private void validateUserData(User user) {
+        if (user == null
+                || user.getLogin() == null
+                || user.getAge() == null
+                || user.getPassword() == null
+                || user.getLogin().length() < MINIMUM_LOGIN_LENGTH
+                || user.getAge() < MINIMUM_USER_AGE
+                || user.getPassword().length() < MINIMUM_PASSWORD_LENGTH) {
+            throw new RuntimeException("Invalid user data!");
+        }
```

Comment:
I think we should throw narrow Exception for better understanding - Invalid login or Invalid age or Invalid password

Reply:
The task says that in case of invalid data, you should throw a RuntimeException. But it is not said that you need to throw for each exception separately